osu.Framework/Localisation/PluralisableString.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+
+#nullable enable
+
+namespace osu.Framework.Localisation
+{
+    /// <summary>
+    /// A string which can display a plural variant of a localised string according to the plural context.
+    /// </summary>
+    public class PluralisableString : TranslatableString
+    {
+        private const char variant_separator = '|';
+
+        /// <summary>
+        /// The plural count to use when applying plural rules.
+        /// </summary>
+        public readonly int Count;
+
+        /// <summary>
+        /// Creates a <see cref="PluralisableString"/> using texts.
+        /// </summary>
+        /// <param name="key">The key for <see cref="LocalisationManager"/> to look up with.</param>
+        /// <param name="fallback">The fallback string to use when no translation can be found.</param>
+        /// <param name="count">The plural count to use when applying plural rules.</param>
+        /// <param name="args">Optional formattable arguments.</param>
+        public PluralisableString(string key, string fallback, int count, params object[] args)
+            : base(key, fallback, args)
+        {
+            Count = count;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="PluralisableString"/> using interpolated string.
+        /// </summary>
+        /// <param name="key">The key for <see cref="LocalisationManager"/> to look up with.</param>
+        /// <param name="interpolation">The interpolated string containing fallback and formattable arguments.</param>
+        /// <param name="count">The plural count to use when applying plural rules.</param>
+        public PluralisableString(string key, FormattableString interpolation, int count)
+            : base(key, interpolation)
+        {
+            Count = count;
+        }
+
+        protected override string GetLocalisedFormat(LocalisationParameters parameters, string format)
+        {
+            string[] variants = format.Split(variant_separator);
+
+            //use the last plural form available for the current locale if the requested form is missing from translated strings.
+            return variants.ElementAtOrDefault(getPluralIndex(parameters)) ?? variants.ElementAt(variants.Length - 1);
```

Comment:
Not sure about using `.ElementAt()` on an array. As far as I can tell [it shouldn't incur allocs](https://github.com/dotnet/runtime/blob/v5.0.11/src/libraries/System.Linq/src/System/Linq/ElementAt.cs) but you may as well write

```suggestion
            // use the last plural form available for the current locale if the requested form is missing from translated strings.
            int pluralIndex = getPluralIndex(parameters);
            return variants[Math.Min(pluralIndex, variants.Length - 1)];
```